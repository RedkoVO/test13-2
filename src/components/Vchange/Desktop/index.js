import React from 'react'
import PropTypes from 'prop-types'
import Select from '@material-ui/core/Select'
import MenuItem from '@material-ui/core/MenuItem'
import withStyles from '@material-ui/core/styles/withStyles'

import Graph from '../components/Graph'
import PricesBuy from '../components/PricesBuy'
import PricesSell from '../components/PricesSell'

import BTCIcon from '../../../assets/images/btc_icon.png'
import USDIcon from '../../../assets/images/usd_icon.png'

import styles from './styles'

const Vchange = ({
  classes,
  pricesBuyData,
  pricesSellData,
  curencySelectFirstValue,
  curencySelectSecondValue,
  handleChangeFirsCurrency,
  handleChangeSecondCurrency
}) => (
  <main>
    <div className={classes.root}>
      <div className={classes.title}>Pair to exchange</div>
      <div className={classes.wrChooseCurrency}>
        <div className={classes.wrCurrency}>
          <Select
            className={classes.selectCurency}
            value={curencySelectFirstValue}
            onChange={e => handleChangeFirsCurrency(e.target.value)}
            IconComponent={props => (
              <i {...props} className={classes.curencyIcon} />
            )}
            inputProps={{
              id: 'curencyDropdownFirst',
              name: 'firstCurency'
            }}
            SelectDisplayProps={{ className: classes.selectField }}
            MenuProps={{ classes: { paper: classes.dropdownStyle } }}
          >
            <MenuItem value={'BTC'} key={1}>
              <div className={classes.curency}>
                <img src={BTCIcon} alt="" />
                BTC
              </div>
            </MenuItem>
            <MenuItem value={'USD'} key={2}>
              <div className={classes.curency}>
                <img src={USDIcon} alt="" />
                USD
              </div>
            </MenuItem>
            <MenuItem value={'EUR'} key={3}>
              <div className={classes.curency}>
                <img src={BTCIcon} alt="" />
                EUR
              </div>
            </MenuItem>
          </Select>
          <div className={classes.balance}>
            Balance: 0.2658 {curencySelectFirstValue}
          </div>
        </div>

        <div className={classes.hyphen}>-</div>

        <div className={classes.wrCurrency}>
          <Select
            className={classes.selectCurency}
            value={curencySelectSecondValue}
            onChange={e => handleChangeSecondCurrency(e.target.value)}
            IconComponent={props => (
              <i {...props} className={classes.curencyIcon} />
            )}
            inputProps={{
              id: 'curencyDropdownSecond',
              name: 'secondCurency'
            }}
            SelectDisplayProps={{ className: classes.selectField }}
            MenuProps={{ classes: { paper: classes.dropdownStyle } }}
          >
            <MenuItem value={'BTC'} key={1}>
              <div className={classes.curency}>
                <img src={BTCIcon} alt="" />
                BTC
              </div>
            </MenuItem>
            <MenuItem value={'USD'} key={2}>
              <div className={classes.curency}>
                <img src={USDIcon} alt="" />
                USD
              </div>
            </MenuItem>
            <MenuItem value={'EUR'} key={3}>
              <div className={classes.curency}>
                <img src={BTCIcon} alt="" />
                EUR
              </div>
            </MenuItem>
          </Select>
          <div className={classes.balance}>
            Balance: 2378.89 {curencySelectSecondValue}
          </div>
        </div>
      </div>

      <Graph />

      <div className={classes.prices}>
        <PricesBuy pricesBuyData={pricesBuyData} />
        <PricesSell pricesSellData={pricesSellData} />
      </div>
    </div>
  </main>
)

Vchange.propTypes = {
  pricesBuyData: PropTypes.array,
  pricesSellData: PropTypes.array,
  curencySelectFirstValue: PropTypes.string,
  curencySelectSecondValue: PropTypes.string,
  handleChangeFirsCurrency: PropTypes.func,
  handleChangeSecondCurrency: PropTypes.func
}

export default withStyles(styles)(Vchange)
