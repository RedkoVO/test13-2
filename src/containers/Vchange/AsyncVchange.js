import Async from '../../components/App/components/Async'

export const AsyncVchange = Async({
  loader: () => import(/* webpackChunkName: "vchange_page" */ './index')
})

export default AsyncVchange
