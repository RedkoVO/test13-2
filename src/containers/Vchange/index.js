import compose from 'recompose/compose'
import { withProps, withHandlers, withState, pure } from 'recompose'

import VchangeDesktop from '../../components/Vchange/Desktop/'

const tmpPricesBuyData = [
  { id: 1, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 2, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 3, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 4, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 5, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 6, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 7, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 8, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 9, price: '3752.22', ammount: '1.0000', total: '3752.2200000' }
]
const tmpPricesSellData = [
  { id: 1, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 2, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 3, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 4, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 5, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 6, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 7, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 8, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 9, price: '3752.22', ammount: '1.0000', total: '3752.2200000' }
]

export default compose(
  withState('curencySelectFirstValue', 'setCurencySelectFirstValue', 'BTC'),
  withState('curencySelectSecondValue', 'setCurencySelectSecondValue', 'USD'),
  withHandlers({
    handleChangeFirsCurrency: ({ setCurencySelectFirstValue }) => value => {
      setCurencySelectFirstValue(value)
    },
    handleChangeSecondCurrency: ({ setCurencySelectSecondValue }) => value => {
      setCurencySelectSecondValue(value)
    }
  }),
  withProps(() => ({
    pricesBuyData: tmpPricesBuyData,
    pricesSellData: tmpPricesSellData
  })),
  pure
)(VchangeDesktop)
