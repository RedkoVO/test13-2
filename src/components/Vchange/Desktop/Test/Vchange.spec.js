import * as React from 'react'
import renderer from 'react-test-renderer'

import Vchange from '../'

const pricesBuyData = [
  { id: 1, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 2, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 3, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 4, price: '3752.22', ammount: '1.0000', total: '3752.2200000' }
]
const pricesSellData = [
  { id: 1, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 2, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 3, price: '3752.22', ammount: '1.0000', total: '3752.2200000' },
  { id: 4, price: '3752.22', ammount: '1.0000', total: '3752.2200000' }
]

it('<Vchange /> renders correctly', () => {
  const tree = renderer
    .create(
      <Vchange
        pricesBuyData={pricesBuyData}
        pricesSellData={pricesSellData}
        curencySelectFirstValue={'BTC'}
        curencySelectSecondValue={'USD'}
        handleChangeFirsCurrency={() => {}}
        handleChangeSecondCurrency={() => {}}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
