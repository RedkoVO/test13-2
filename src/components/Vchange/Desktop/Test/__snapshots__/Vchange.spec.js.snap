// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vchange /> renders correctly 1`] = `
<main>
  <div
    className="Vchange-root-1"
  >
    <div
      className="Vchange-title-2"
    >
      Pair to exchange
    </div>
    <div
      className="Vchange-wrChooseCurrency-3"
    >
      <div
        className="Vchange-wrCurrency-5"
      >
        <div
          className="MuiInputBase-root-33 MuiInput-root-20 MuiInput-underline-24 Vchange-selectCurency-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root-13"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="Vchange-selectField-9"
              id="select-firstCurency"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="Vchange-curency-11"
              >
                <img
                  alt=""
                  src="btc_icon.png"
                />
                BTC
              </div>
            </div>
            <input
              id="curencyDropdownFirst"
              name="firstCurency"
              type="hidden"
              value="BTC"
            />
            <i
              className="Vchange-curencyIcon-10"
            />
          </div>
        </div>
        <div
          className="Vchange-balance-6"
        >
          Balance: 0.2658 
          BTC
        </div>
      </div>
      <div
        className="Vchange-hyphen-4"
      >
        -
      </div>
      <div
        className="Vchange-wrCurrency-5"
      >
        <div
          className="MuiInputBase-root-33 MuiInput-root-20 MuiInput-underline-24 Vchange-selectCurency-7"
          onClick={[Function]}
        >
          <div
            className="MuiSelect-root-13"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              className="Vchange-selectField-9"
              id="select-secondCurency"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="Vchange-curency-11"
              >
                <img
                  alt=""
                  src="usd_icon.png"
                />
                USD
              </div>
            </div>
            <input
              id="curencyDropdownSecond"
              name="secondCurency"
              type="hidden"
              value="USD"
            />
            <i
              className="Vchange-curencyIcon-10"
            />
          </div>
        </div>
        <div
          className="Vchange-balance-6"
        >
          Balance: 2378.89 
          USD
        </div>
      </div>
    </div>
    <div
      className="Graph-root-54"
    >
      <img
        alt=""
        src="graphTMP.png"
      />
    </div>
    <div
      className="Vchange-prices-12"
    >
      <div
        className="PricesBuy-root-55"
      >
        <div
          className="PricesBuy-title-56"
        >
          <span>
            Buy
          </span>
           BTC: Prices
        </div>
        <table
          cellSpacing="0"
          className="PricesBuy-wrTable-57"
        >
          <tbody>
            <tr>
              <th>
                Price (USDT)
              </th>
              <th>
                Ammount (BTC)
              </th>
              <th>
                Total (USDT)
              </th>
            </tr>
            <tr
              className="PricesBuy-wrItem-58"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesBuy-wrItem-58"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesBuy-wrItem-58"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesBuy-wrItem-58"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="PricesSell-root-59"
      >
        <div
          className="PricesSell-title-60"
        >
          <span>
            Sell
          </span>
           BTC: Prices
        </div>
        <table
          cellSpacing="0"
          className="PricesSell-wrTable-61"
        >
          <tbody>
            <tr>
              <th>
                Price (USDT)
              </th>
              <th>
                Ammount (BTC)
              </th>
              <th>
                Total (USDT)
              </th>
            </tr>
            <tr
              className="PricesSell-wrItem-62"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesSell-wrItem-62"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesSell-wrItem-62"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
            <tr
              className="PricesSell-wrItem-62"
            >
              <td>
                3752.22
              </td>
              <td>
                1.0000
              </td>
              <td>
                3752.2200000
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</main>
`;
